spec/models/submission_rule_spec.rb

```diff
@@ -602,4 +620,109 @@
       end
     end
   end
+
+  #FAILING, GracePeriodSubmissionRule factory not properly initialized
+  context 'Grace period ids' do
+    before(:each) do
+    submission_rule = create(:GracePeriodSubmissionRule)
+    @sub_rule_id = @submission_rule.id
+
+    # Randomly create five periods for this SubmissionRule (ids unsorted):
+    @period = create(:Period, submission_rule_id: @sub_rule_id)
+    first_period_id = @period.id
+
+    # Create four other periods
+      for i in (1..4) do
+        @period = create(:Period, id: first_period_id + i, submission_rule_id: @sub_rule_id)
+      end
+    end
+
+    it 'sort in ascending order' do
+      # Loop through periods for this SubmissionRule and verify the ids are
+      # sorted in ascending order
+      previous_id = @submission_rule.periods[0][:id]
+      for i in (1..4) do
+        expect(@submission_rule.periods[i][:id]).to be > previous_id
+        previous_id = @submission_rule.periods[i][:id]
+      end
+    end
+  end
+
+  #FAILING, PenaltyPeriodSubmissionRule factory not properly initialized
+  context 'Penalty period ids' do
+    before(:each) do
+    # Create SubmissionRule with default type 'PenaltyPeriodSubmissionRule'
+    @submission_rule = create(:PenaltyPeriodSubmissionRule)
+    sub_rule_id = @submission_rule.id
+
+    # Randomly create five periods for this SubmissionRule (ids unsorted):
+    @period = create(:Period, submission_rule_id: sub_rule_id)
+    first_period_id = @period.id
+
+    # Create four other periods
+    for i in (1..4) do
+      @period = create(:Period, id: first_period_id + i, submission_rule_id: @sub_rule_id)
+    end
+    end
+
+    it 'should sort in ascending order' do
+      # Loop through periods for this SubmissionRule and verify the ids are sorted in ascending order
+      previous_id = @submission_rule.periods[0][:id]
+      for i in (1..4) do
+        expect(@submission_rule.periods[i][:id]).to be > previous_id
+        previous_id = @submission_rule.periods[i][:id]
+      end
+    end
```

Comment:
Please use one `expect` statement per `it` context. See https://stackoverflow.com/a/34858160/5992438 for suggestions of how to create `it` contexts in a loop